astropy/modeling/core.py

```diff
@@ -1212,8 +1250,24 @@ def render(self, out=None, coords=None):
 
     @property
     def input_units(self):
```

Comment:
The downside with implementing this like this is that if you want to change this in a subclass in Python 2 compatible land, you need to do a super call to get this optional behaviour.